opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -0,0 +1,299 @@
+      FormatNames.DATE_HOUR_MINUTE_SECOND_FRACTION,
+      FormatNames.DATE_HOUR_MINUTE_SECOND_MILLIS,
+      FormatNames.STRICT_DATE_HOUR_MINUTE_SECOND_MILLIS,
+      FormatNames.DATE_HOUR_MINUTE_SECOND,
+      FormatNames.STRICT_DATE_HOUR_MINUTE_SECOND,
+      FormatNames.DATE_HOUR_MINUTE,
+      FormatNames.STRICT_DATE_HOUR_MINUTE,
+      FormatNames.DATE_HOUR,
+      FormatNames.STRICT_DATE_HOUR,
+      FormatNames.ORDINAL_DATE_TIME,
+      FormatNames.STRICT_ORDINAL_DATE_TIME,
+      FormatNames.ORDINAL_DATE_TIME_NO_MILLIS,
+      FormatNames.STRICT_ORDINAL_DATE_TIME_NO_MILLIS,
+      FormatNames.WEEK_DATE_TIME,
+      FormatNames.STRICT_WEEK_DATE_TIME,
+      FormatNames.WEEK_DATE_TIME_NO_MILLIS,
+      FormatNames.STRICT_WEEK_DATE_TIME_NO_MILLIS
+  );
+
+  // list of named formats that only support year/month/day
+  public static final List<FormatNames> SUPPORTED_NAMED_DATE_FORMATS = List.of(
+      FormatNames.BASIC_DATE,
+      FormatNames.BASIC_ORDINAL_DATE,
+      FormatNames.DATE,
+      FormatNames.STRICT_DATE,
+      FormatNames.YEAR_MONTH_DAY,
+      FormatNames.STRICT_YEAR_MONTH_DAY,
+      FormatNames.YEAR_MONTH,
+      FormatNames.STRICT_YEAR_MONTH,
+      FormatNames.YEAR,
+      FormatNames.STRICT_YEAR,
+      FormatNames.ORDINAL_DATE,
+      FormatNames.STRICT_ORDINAL_DATE,
+      FormatNames.WEEK_DATE,
+      FormatNames.STRICT_WEEK_DATE,
+      FormatNames.WEEKYEAR_WEEK_DAY,
+      FormatNames.STRICT_WEEKYEAR_WEEK_DAY,
+      FormatNames.WEEK_YEAR,
+      FormatNames.WEEK_YEAR_WEEK,
+      FormatNames.STRICT_WEEKYEAR_WEEK,
+      FormatNames.WEEKYEAR,
+      FormatNames.STRICT_WEEKYEAR
+  );
+
+  // list of named formats that only support hour/minute/second
+  public static final List<FormatNames> SUPPORTED_NAMED_TIME_FORMATS = List.of(
+      FormatNames.BASIC_TIME,
+      FormatNames.BASIC_TIME_NO_MILLIS,
+      FormatNames.BASIC_T_TIME,
+      FormatNames.BASIC_T_TIME_NO_MILLIS,
+      FormatNames.TIME,
+      FormatNames.STRICT_TIME,
+      FormatNames.TIME_NO_MILLIS,
+      FormatNames.STRICT_TIME_NO_MILLIS,
+      FormatNames.HOUR_MINUTE_SECOND_FRACTION,
+      FormatNames.STRICT_HOUR_MINUTE_SECOND_FRACTION,
+      FormatNames.HOUR_MINUTE_SECOND_MILLIS,
+      FormatNames.STRICT_HOUR_MINUTE_SECOND_MILLIS,
+      FormatNames.HOUR_MINUTE_SECOND,
+      FormatNames.STRICT_HOUR_MINUTE_SECOND,
+      FormatNames.HOUR_MINUTE,
+      FormatNames.STRICT_HOUR_MINUTE,
+      FormatNames.HOUR,
+      FormatNames.STRICT_HOUR,
+      FormatNames.T_TIME,
+      FormatNames.STRICT_T_TIME,
+      FormatNames.T_TIME_NO_MILLIS,
+      FormatNames.STRICT_T_TIME_NO_MILLIS
+  );
+
+  @EqualsAndHashCode.Exclude
+  String formatString;
+
+  private OpenSearchDateType() {
+    super(MappingType.Date);
+    this.formatString = "";
+  }
+
+  private OpenSearchDateType(ExprCoreType exprCoreType) {
+    super(MappingType.Date);
+    this.formatString = "";
+    this.exprCoreType = exprCoreType;
```

Comment:
You can call one constructor from another one:
```suggestion
    this();
    this.exprCoreType = exprCoreType;
```
And the same below.

Reply:
Done in https://github.com/Bit-Quill/opensearch-project-sql/pull/268/commits/58a8f193ffe1407665734d55a60e53ff9bc97aa8